src/AtomsMCMoves.jl

```diff
@@ -0,0 +1,368 @@
+            end
+        else
+            error("Unsupported boundary condition: $(pbc[i]).")
+        end
+    end
+    pos = SVector{3,T}(new_pos)
+    return pos
+end
+
+"""
+    mean_sq_displacement(at::AtomWalker, at_orig::AtomWalker)
+
+Calculate the mean squared displacement before and after random walk(s).
+
+# Arguments
+- `at::AtomWalker`: The current `AtomWalker` after the random walk.
+- `at_orig::AtomWalker`: The original `AtomWalker` before the random walk.
+
+# Returns
+- `distsq::typeof(0.0u"Å"^2)`: The mean squared displacement of all free particles.
+
+"""
+function mean_sq_displacement(at::AtomWalker, at_orig::AtomWalker)
+    distsq::typeof(0.0u"Å"^2) = 0.0u"Å"^2
+    frozen = at.num_frozen_part
+    for i in (frozen+1):length(at.configuration)
+        dist::typeof(0.0u"Å") = pbc_dist(at.configuration.position[i], at_orig.configuration.position[i], at.configuration)
+        distsq += dist^2
+    end
+    return distsq/(length(at.configuration)-frozen)
+end
+
+
+"""
+    single_atom_random_walk!(pos::SVector{3,T}, step_size::Float64) where T
+
+Perform a single atom random walk by updating the position `pos` in each direction by a random amount.
+The `step_size` determines the maximum distance the atom can move in any direction.
+
+# Arguments
+- `pos::SVector{3,T}`: The current position of the atom as a 3D vector.
+- `step_size::Float64`: The maximum distance the atom can move in any direction.
+
+# Returns
+- `pos`: The updated position of the atom.
+
+"""
+function single_atom_random_walk!(pos::SVector{3,T}, step_size::Float64) where T
+    (dx, dy, dz) = (rand(Uniform(-step_size,step_size)) for _ in 1:3) .* unit(T)
+    pos = pos .+ (dx, dy, dz)
+    return pos 
+end
+
+
+"""
+    MC_random_walk!(n_steps::Int, at::AtomWalker, lj::LJParameters, step_size::Float64, emax::typeof(0.0u"eV"))
+
+Perform a Monte Carlo random walk on the atomic/molecular system.
+
+# Arguments
+- `n_steps::Int`: The number of Monte Carlo steps to perform.
+- `at::AtomWalker`: The walker to perform the random walk on.
+- `lj::LJParameters`: The Lennard-Jones potential parameters.
+- `step_size::Float64`: The maximum distance an atom can move in any direction.
+- `emax::typeof(0.0u"eV")`: The maximum energy allowed for accepting a move.
+
+# Returns
+- `accept_this_walker::Bool`: Whether the walker is accepted or not.
+- `accept_rate::Float64`: The acceptance rate of the random walk.
+- `at::AtomWalker`: The updated walker.
+
+"""
+function MC_random_walk!(
+                    n_steps::Int, 
+                    at::AtomWalker, 
+                    lj::LJParameters, 
+                    step_size::Float64, 
+                    emax::typeof(0.0u"eV")
+                    )
+    n_accept = 0
+    accept_this_walker = false
+    for i_mc_step in 1:n_steps
+        config = at.configuration
+        frozen = at.num_frozen_part
+        e_shift = at.energy_frozen_part
+        i_at = rand((frozen+1):length(config))
```

Comment:
@yangmr04, is there a benefit to doing single-atom compared to multiple-atom random walks?